client/src/App.js

```diff
@@ -10,15 +10,32 @@ import './App.scss'
 class App extends Component {
   state = {}
 
-  async componentDidMount() {
+  loadReports = async () => {
     const tags = await getTags()
-    this.setState({tags})
+    const reportData = await getReportData()
+    const reportByTs = Object.create(null)
+    reportData.reports.forEach((report) => {reportByTs[report.ts] = report})
+    const reportsByTag = Object.create(null)
+    tags.forEach(
+      ({tag, reports}) =>
+        {reportsByTag[tag] = reports.map((ts) => reportByTs[ts])}
+    )
+    reportData.reportsByTag = reportsByTag
```

Comment:
I think doing this will be easier on the server, I will explain it in the other comment.